mealie/core/dependencies/dependencies.py

```diff
@@ -205,9 +205,13 @@ def validate_recipe_token(token: str | None = None) -> str:
     return slug
 
 
-async def temporary_zip_path() -> AsyncGenerator[Path, None]:
+def temporary_zip_path() -> Path:
```

Comment:
Can we change this to `get_temporary_zip_path` and not use it as a dependency? More info below

Reply:
Yep.  Though I should note that `temporary_dir` is used as a route parameter for `POST /users/{id}/image` and `GET /recipe/{slug}/exports`.  I'll have to figure out exactly what `Depends` does to figure out if I can remove it from the parameter list and still have it clean up after itself.

Reply:
It looks like only the async + yield does the cleanup, so currently `temporary_dir` doesn't do any cleanup with the new behavior.

Reply:
That's what I'm seeing.  I think I'm going to have to push the cleanup code into those two routes.